core/server/api_container/server/service_network/default_service_network.go

```diff
@@ -1551,20 +1580,35 @@ func (network *DefaultServiceNetwork) copyFilesFromServiceUnlocked(ctx context.C
 		return "", stacktrace.Propagate(err, "An error occurred getting the files artifact store")
 	}
 
+	existingFilesArtifactUuid, _, _, filesArtifactAlreadyExists, err := store.GetFile(artifactName)
+	if err != nil {
+		return "", stacktrace.Propagate(err, "An unexpected error occurred checking for file artifact '%s' existence in the store", artifactName)
+	}
+
 	pipeReader, pipeWriter := io.Pipe()
 	defer pipeReader.Close()
```

Comment:
out of the scope, but just to check, I'm seeing that this reader is closed in the main routine and in the subroutine, is this ok?